projects/components/src/stepper/stepper.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  EventEmitter,
+  Input,
+  Output,
+  QueryList
+} from '@angular/core';
+import { MatStepper } from '@angular/material/stepper';
+import { queryListAndChanges$ } from '@hypertrace/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ButtonRole, ButtonStyle } from '../button/button';
+import { IconSize } from '../icon/icon-size';
+import { StepperTabComponent } from './stepper-tab.component';
+
+@Component({
+  selector: 'ht-stepper',
+  template: `
+    <div class="stepper" *htLoadAsync="this.steps$ as steps">
+      <mat-stepper [orientation]="this.orientation" [linear]="this.isLinear" #stepper>
+        <ng-container *ngFor="let step of steps">
+          <mat-step [completed]="step.completed" [stepControl]="step.stepControl" [optional]="step.optional">
+            <ng-template matStepLabel>
+              <ht-label class="header-label" [label]="step.label"></ht-label>
+            </ng-template>
+            <ng-container *ngTemplateOutlet="step.content"></ng-container>
+            <ng-template matStepperIcon="edit">
+              <ht-icon [icon]="step.icon" size="${IconSize.ExtraSmall}"> </ht-icon>
+            </ng-template>
+          </mat-step>
+        </ng-container>
+      </mat-stepper>
+      <div class="action-buttons">
+        <ht-button
+          class="cancel"
+          label="Cancel"
+          (click)="this.onClickCancel()"
+          role="${ButtonRole.Tertiary}"
+        ></ht-button>
+        <div class="back-next">
+          <ht-button
+            class="back"
+            *ngIf="stepper.selectedIndex !== 0"
+            display="${ButtonStyle.Outlined}"
+            label="Back"
+            (click)="this.onClickBack(stepper)"
+          ></ht-button>
+          <ht-button
+            class="next"
+            *ngIf="stepper.selectedIndex !== steps.length - 1"
+            label="Next"
+            (click)="this.onClickNext(stepper)"
+            [disabled]="
+              this.isNextDisabled
+                | htMemoize
+                  : this.isLinear
+                  : stepper.steps.get(stepper.selectedIndex)?.completed
+                  : stepper.steps.get(stepper.selectedIndex)?.stepControl?.valid
+            "
+          ></ht-button>
+          <ht-button
+            class="submit"
+            *ngIf="stepper.selectedIndex === steps.length - 1"
+            label="Submit"
+            (click)="this.onClickSubmit()"
+            [disabled]="
+              this.isSubmitDisabled
+                | htMemoize
+                  : stepper.steps.get(stepper.selectedIndex)?.completed
+                  : stepper.steps.get(stepper.selectedIndex)?.stepControl?.valid
```

Comment:
you don't need two inputs to the method here. what you need here is the object. its the same object accessed for both the arguments

Reply:
Agreed. Updated branch.